internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource.go

```diff
@@ -180,9 +201,14 @@ func resourceDataFactoryLinkedServiceBlobStorageCreateUpdate(d *pluginsdk.Resour
 	}
 
 	if v, ok := d.GetOk("sas_uri"); ok {
-		blobStorageProperties.SasURI = &datafactory.SecureString{
-			Value: utils.String(v.(string)),
-			Type:  datafactory.TypeSecureString,
+		if sasToken, ok := d.GetOk("key_vault_sas_token"); ok {
+			blobStorageProperties.SasURI = utils.String(v.(string))
+			blobStorageProperties.SasToken = expandAzureKeyVaultSecretReference(sasToken.([]interface{}))
+		} else {
+			blobStorageProperties.SasURI = &datafactory.SecureString{
+				Value: utils.String(v.(string)),
+				Type:  datafactory.TypeSecureString,
+			}
```

Comment:
You need to set those properties back in `Read()` method as well, otherwise you'll always end up with a diff, as you can see in the acceptance test output comment.